pkgs/development/tools/build-managers/cmake/setup-hook.sh

```diff
@@ -7,7 +7,7 @@ fixCmakeFiles() {
     echo "fixing cmake files..."
     find "$1" \( -type f -name "*.cmake" -o -name "*.cmake.in" -o -name CMakeLists.txt \) -print |
         while read fn; do
-            sed -e 's^/usr\([ /]\|$\)^/var/empty\1^g' -e 's^/opt\([ /]\|$\)^/var/empty\1^g' < "$fn" > "$fn.tmp"
+            sed -e '/\${\(_CMAKE_OSX_SYSROOT_PATH\|sdk_path\)}\/usr/! s^/usr\([ /]\|$\)^/var/empty\1^g' -e 's^/opt\([ /]\|$\)^/var/empty\1^g' < "$fn" > "$fn.tmp"
```

Comment:
Can we add a comment to explain this rather complex sed? Maybe something like we replace X but ignore Y because it breaks on darwin.